api_test.go

```diff
@@ -208,3 +211,52 @@ func TestHttpApi(t *testing.T) {
 
 	tests.TestApi(newNodeProvider(ctx))(t)
 }
+
+func Test_NewURLApiWithClient(t *testing.T) {
+	t.Skip()
+	var (
+		url                 = "127.0.0.1:65501"
```

Comment:
I don't really like using fixed ports in tests, but we can live with that (feel free to ignore)

Reply:
I wrote this before realizing that `httpest.NewServer` will use a random port number

```
	ts := httptest.NewServer(
		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			val := r.Header.Get(headerToTest)
			if val == expectedHeaderValue {
				w.WriteHeader(400)
				return
			}
			w.WriteHeader(200)
		}),
	)
```

We can then access the IP+Port from `ts.URL` so I'll go ahead and move to that

Reply:
Sounds good